engine/consensus/matching/engine.go

```diff
@@ -158,6 +200,18 @@ func (e *Engine) loop() {
 	}
 }
 
+func (e *Engine) processFinalizationEvent() error {
+	finalized, err := e.state.Final().Head()
+	if err != nil {
+		return fmt.Errorf("could not retrieve last finalized block: %w", err)
+	}
+	err = e.core.ProcessFinalizedBlock(finalized.ID())
```

Comment:
I would suggest to rename method, otherwise, it sounds as if `core.ProcessFinalizedBlock` is expecting to be feed with every single finalized block, but actually we only need to provide the latest
```suggestion
	err = e.core.ProcessFinalizedBlock(finalized.ID())
```